File: backend/app/schemas/session.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.conversation_session import StudentType


class SessionCreateRequest(BaseModel):
    student_type: StudentType | None = None
    semester: str | None = None
    college: str | None = None
    department: str | None = None


class SessionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    student_type: StudentType | None
    semester: str | None
    college: str | None
    department: str | None
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/session.ts
import type { StudentType } from "./chat";

export interface SessionCreateRequest {
  student_type?: StudentType | null;
  semester?: string | null;
  college?: string | null;
  department?: string | null;
}

export interface SessionResponse {
  id: string;
  student_type: StudentType | null;
  semester: string | null;
  college: string | null;
  department: string | null;
  created_at: string;
  updated_at: string;
}
